### PR TITLE
Add extra classes for interactives

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?

Adding extra class names to support Visuals in customising interactives. These class names represent a contract that we should try to maintain.

## Why?

To support the customisation of interactives and thereby reducing the number of dcr opt-outs for new interactives.

### Before vs After

N/A - no UI changes

### CODE REVIEW

**Review by nicl — APPROVED** [review]
Not necessarily part of this PR, but I wonder if we should add some tests for the presence of these classes in interactive pages going forward?

File: docs/contracts/003-extra-classes-for-interactives.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Extra Classes For Interactives
```
Comment (jlieb10) [inline docs/contracts/003-extra-classes-for-interactives.md:1]: Thank you for the documentation!

File: src/web/components/ArticleMeta.tsx [hunk 2]
```diff
@@ -264,7 +265,10 @@ export const ArticleMeta = ({
 
 	const showAvatar = onlyOneContributor && shouldShowAvatar(format);
 	return (
-		<div css={metaContainer(format)}>
+		<div
+			className={interactiveLegacyClasses.metaContainer}
```
Comment (jlieb10) [inline src/web/components/ArticleMeta.tsx:269]: I'm not sure how className and CSS work together when both specified - as the css compiles down into a classname itself.

**Review by jlieb10 — APPROVED** [review]
Nice work!

File: src/web/components/ArticleMeta.tsx (hunk 2 above)
Comment (DavidLawes) [reply src/web/components/ArticleMeta.tsx:269]: ah, when I tested the static class name was included as well as the auto-generated one, eg:

Screenshot 2021-08-16 at 14 08 58

does that answer your question?

File: src/model/json-schema.json [hunk 3]
```diff
@@ -3628,10 +3628,7 @@
                     }
                 },
                 "abTests": {
-                    "type": "object",
-                    "additionalProperties": {
-                        "type": "string"
-                    }
+                    "$ref": "#/definitions/Record<string,\"control\"|\"variant\">"
```
Comment (DavidLawes) [inline src/model/json-schema.json:3631]: I think this is ok? Looks like recently the type for this property was update https://github.com/guardian/dotcom-rendering/pull/3276/files

File: src/web/components/ArticleMeta.tsx (hunk 2 above)
Comment (jlieb10) [reply src/web/components/ArticleMeta.tsx]: yes - thanks!

File: src/model/json-schema.json (hunk 3 above)
Comment (jlieb10) [reply src/model/json-schema.json:3631]: i'm not sure I understand these updates

**Comment by jlieb10** [comment]
@DavidLawes the new tests look great and can go ahead - but I just lmk about the schema file when you get a chance. Also, I think you might need to merge `main` in since the code freeze because of the conflicts

File: src/model/json-schema.json (hunk 3 above)
Comment (DavidLawes) [reply src/model/json-schema.json]: changes to this file are now ignored :)

**Pull request merged**